### PR TITLE
Allowing the bot to re-run cancelled tests

### PR DESCRIPTION
**Pull request by mkaddani**
Allow the bot to rerun Cancelled tests, see  #1474.

Tested on https://github.com/EmilyBourne/pyccel

### CODE REVIEW

**Review by EniddeallA — APPROVED** [review]
Looks good!

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ All notable changes to this project will be documented in this file.
 -   #682 : Wrong data layout when copying a slice of an array.
 -   #1453 : Fix error-level developer mode output.
 -   \[INTERNALS\] Fix string base class selection.
+-   #1502 : Allowing the bot to re-run cancelled tests.
```
Comment (EmilyBourne) [inline CHANGELOG.md:22]: Changes to the bot do not need to appear in the CHANGELOG. The CHANGELOG is about changes in the code. It serves as a reference for users to decide which version of Pyccel they need to use. Changes to the bot do not affect users and developers can't choose a version of the bot to use for their PRs so this information is not useful here

**Review by EmilyBourne — CHANGES_REQUESTED** [review]
Looks good. I just have 1 minor comment

File: CHANGELOG.md (hunk 1 above)
Comment (mkaddani) [reply CHANGELOG.md]: Thank you, for your review; Fixed.

**Review by EmilyBourne — APPROVED** [review]

**Review by yguclu — APPROVED** [review]

**Comment by EmilyBourne** [comment]
> /bot run pr_tests

The tests ran, they failed. You cannot run them all again without adding a new commit. You also shouldn't run them all again given that most passed. Looking at the error from the one failing workflow it looks like a fluke so I have set it to rerun

**Pull request merged**